### PR TITLE
more functions

### PR DESCRIPTION
**Pull request by bheni**
Signed-off-by: Brian Hendriks <brian@liquidata.co>

### CODE REVIEW

File: sql/expression/function/function_test.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
+package function
```
Comment (bheni) [inline sql/expression/function/function_test.go:1]: This file has a bunch of helper classes for setting up and running tests.

File: sql/expression/function/string.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	"fmt"
+	"github.com/liquidata-inc/go-mysql-server/sql"
+	"github.com/shopspring/decimal"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+)
+
+// AsciiFunc implements the sql function "ascii" which returns the
```
Comment (zachmu) [inline sql/expression/function/string.go:14]: don't leave me hanging

File: sql/expression/function/string.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+func hexChar(b byte) byte {
+	if b > 9 {
+		return b - 10 + byte('A')
+	}
+
+	return b + byte('0')
+}
+
+// MySQL expects the 64 bit 2s compliment representation for negative integer values. Typical methods for converting a
```
Comment (zachmu) [inline sql/expression/function/string.go:46]: complement

File: sql/expression/function/string.go [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+		n := int64(f)
+		return hexForNegativeInt64(n), nil
+	}
+
+	f += 0.5
+	n := uint64(f)
+	return fmt.Sprintf("%X", n), nil
+}
+
+func HexFunc(_ *sql.Context, arg interface{}) (interface{}, error) {
```
Comment (zachmu) [inline sql/expression/function/string.go:73]: Method doc

File: sql/expression/function/string.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+	s := ""
+	for i := 7; i >= 0; i-- {
+		s += strconv.FormatInt(int64(bytes[i]), 2)
+	}
+
+	return s
+}
+
+func BinFunc(_ *sql.Context, arg interface{}) (interface{}, error) {
```
Comment (zachmu) [inline sql/expression/function/string.go:184]: Method doc

File: sql/expression/function/string.go [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+		if n.(int64) < 0 {
+			return binForNegativeInt64(n.(int64)), nil
+		} else {
+			return strconv.FormatInt(n.(int64), 2), nil
+		}
+	}
+}
+
+func BitLengthFunc(_ *sql.Context, arg interface{}) (interface{}, error) {
```
Comment (zachmu) [inline sql/expression/function/string.go:206]: Method doc

**Pull request merged**